### PR TITLE
#1216 show diagnostic signs on all folders

### PR DESCRIPTION
**Pull request by alex-courtis**
Now consistent with git behaviour: show diagnostic on all folders

Feature that can use this: #1245

See #1216

### CODE REVIEW

File: lua/nvim-tree/utils.lua [hunk 1]
```diff
@@ -123,6 +123,29 @@ function M.find_node(nodes, fn)
   return node, i
 end
 
+-- return visible nodes indexed by line
+-- @param nodes_all list of node
+-- @param line_start first index
+---@return table
+function M.get_nodes_by_line(nodes_all, line_start)
```
Comment (alex-courtis) [inline lua/nvim-tree/utils.lua:130]: refactored to avoid circular dependencies

File: lua/nvim-tree/diagnostics.lua [hunk 2]
```diff
@@ -140,18 +140,17 @@ function M.update()
     local bufpath = utils.canonical_path(bufname)
     log.line("diagnostics", " bufpath '%s' severity %d", bufpath, severity)
     if 0 < severity and severity < 5 then
-      local node, line = utils.find_node(core.get_explorer().nodes, function(node)
+      local nodes_by_line = utils.get_nodes_by_line(core.get_explorer().nodes, core.get_nodes_starting_line())
```
Comment (alex-courtis) [inline lua/nvim-tree/diagnostics.lua:143]: These are the actual changes.

**Review by kyazdani42 — APPROVED** [review]
Thanks :)

**Comment by kyazdani42** [comment]
please rebase :)

**Pull request merged**